app/controllers/sessions_controller.rb

```diff
@@ -2,23 +2,15 @@
 
 # SessionsController
 class SessionsController < ApplicationController
-  def new; end
-
-  def load_user
-    @user = User.find_by(email: params.dig(:session, :email)&.downcase)
-    return if @user
+  before_action :load_user, only: :create
 
-    flash.now[:danger] = t("layouts.messages.user_not_found")
-    render :new, status: :unprocessable_entity
-  end
+  def new; end
 
   def create
-    load_user
-    return if performed? # Check if render or redirect has already been called
-
-    if @user&.authenticate params.dig(:session, :password)
+    if @user.try(:authenticate, params.dig(:session, :password))
```

Comment:
```suggestion
    if @user.authenticate params.dig(:session, :password)
```
Sau khi load_user rồi thì khi vào tới đây, @user không thể nil được
mô tả một chút về `try`

Reply:
**try** cho phép chúng ta truyền vào một phương thức trong 1 đối tượng và kèm thêm các đối số được truyền vào nếu cần, nếu phương thức tồn tại thì sẽ được gọi và sẽ trả về kết quả, nếu không thì sẽ trả về nil mà không raise exception tránh crash cho website.